Doc/contributing.rst

```diff
@@ -0,0 +1,269 @@
+==============================
+
+If you're used to open-source Python development with Git, here's the gist:
+
+* ``git clone https://github.com/python-ldap/python-ldap``
+* Use GitHub for `the bug tracker`_ and pull requests.
+* Run tests with `tox`_; ignore Python interpreters you don't have locally.
+
+.. _the bug tracker: https://github.com/python-ldap/python-ldap/issues
+.. _tox: https://tox.readthedocs.io/en/latest/
+
+Or, if you prefer to avoid closed-source services:
+
+* ``git clone https://pagure.io/python-ldap``
+* Send bug reports and patches to the mailing list.
+* Run tests with `tox`_; ignore Python interpreters you don't have locally.
+* Read the documentation directly at `Read the Docs`_.
+
+.. _Read the Docs: http://python-ldap.readthedocs.io/
+
+If you're new to some aspect of the project, you're welcome to use (or adapt)
+the workflow below.
+
+
+Sample workflow
+---------------
+
+We assume that, as a user of python-ldap you're not new to software
+development in general, so these instructions are terse.
+If you need additional detail, please do ask on the mailing list.
+
+.. note::
+
+    The following instructions are for Linux.
+    If you can translate them to another system, please contribute your
+    translation!
+
+
+Install `Git`_ and `tox`_.
+
+Clone the repository::
+
+    $ git clone https://github.com/python-ldap/python-ldap
+    $ cd python-ldap
+
+Create a `virtual environment`_ to ensure you in-development python-ldap won't
+affect the rest of your system::
+
+    $ python3 -m venv __venv__
+
+(For Python 2, install `virtualenv`_ and use it instead of ``python3 -m venv``.)
+
+.. _git: https://git-scm.com/
+.. _virtual environment: https://docs.python.org/3/library/venv.html
+.. _virtualenv: https://virtualenv.pypa.io/en/stable/
+
+Activate the virtual environment::
+
+    $ source __venv__/bin/activate
+
+Install python-ldap to it in `editable mode`_::
+
+    (__venv__)$ python -m pip install -e .
+
+This way, importing a Python module from python-ldap will directly
+use the code from your source tree.
+If you change C code, you will still need to recompile
+(using the ``pip install`` command again).
+
+.. _editable mode: https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs
+
+Change the code as desired.
+
+
+To run tests, install and run `tox`_::
+
+    (__venv__)$ python -m pip install tox
+    (__venv__)$ tox
+
+This tries to run tests on all supported versions of Python.
+If you don't have some of them installed, you will get an
+``InterpreterNotFound`` error, which you can ignore::
+
+    ERROR:   py37: InterpreterNotFound: python3.7
+
+To run a subset of test environments, run for example::
+
+    (__venv__)$ tox -e py27,py36
+
+In addition to ``pyXY`` environments, we have extra environments
+for checking things independent of the Python version:
+
+* ``doc`` checks syntax and spelling of the documentation
+* ``coverage`` generates a test coverage report
```

Comment:
* ``coverage-report`` generates a test coverage report for Python code. It must be used as last test env, e.g. ``tox -e py27,py36,coverage-report``.

Reply:
Thanks for the catch!